frontend/src/metabase/containers/ItemPicker/ItemPickerView.tsx

```diff
@@ -74,13 +78,33 @@ function ItemPickerView({
 
   const isPickingNotCollection = models.some(model => model !== "collection");
 
+  // We have the searchString here to act as the middle-man for different search
+  // behaviors. i.e. with the debounce, we can use the useDebouncedValue hook,
+  // or with a keypress we can only set the parent's search text on "Enter"
+  const [searchText, setSearchText] = useState(searchString);
```

Comment:
Could we make `SearchInput` a "controlled" component by passing `searchValue` to set its value? Since the component is re-rendered there would be no performance difference, but the controlled component approach would make it much more straightforward and the comment here won't be needed.